kvbc/src/kvbc_adapter/v4blockchain/app_state_adapter.cpp

```diff
@@ -41,9 +42,12 @@ bool AppStateAdapter::getPrevDigestFromBlock(uint64_t blockId,
                                              bftEngine::bcst::StateTransferDigest *outPrevBlockDigest) const {
   ConcordAssert(blockId > 0);
   const auto parent_digest = kvbc_->parentDigest(blockId);
-  static_assert(parent_digest.size() == DIGEST_SIZE);
   static_assert(sizeof(bftEngine::bcst::StateTransferDigest) == DIGEST_SIZE);
-  std::memcpy(outPrevBlockDigest, parent_digest.data(), DIGEST_SIZE);
+  if (parent_digest == std::nullopt) {
+    return false;
```

Comment:
What state transfer does on a false response?

Reply:
@cloudnoize in most cascalls it will terminate, since it should have never reached this state.  This function is called from multiple places in ST code.